File: leetsync/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int size = nums.length;
        int[] result = new int[size];
        int left = 0;
        int right = size - 1;
        for (int i = 0; i < size; i++) {
            if (nums[i] < pivot) {
                result[left++] = nums[i];
            }
            if (nums[size - 1 - i] > pivot) {
                result[right--] = nums[size - 1 - i];
            }
        }
        while (left <= right) {
            result[left++] = pivot;
            result[right--] = pivot;
        }
        return result;
    }
}

File: leetsync/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.go
func pivotArray(nums []int, pivot int) []int {
	n := len(nums)
	result := make([]int, n)
	left, right := 0, n-1
	i, j := 0, n-1
	for i < n && j >= 0 {
		if nums[i] < pivot {
			result[left] = nums[i]
			left++
		}
		if nums[j] > pivot {
			result[right] = nums[j]
			right--
		}
		i++
		j--
	}
	// Fill pivot in the remaining positions
	for left <= right {
		result[left] = pivot
		left++
	}
	return result
}